animation_nodes/nodes/vector/vector_noise.py

```diff
@@ -1,27 +1,277 @@
+                self.newOutput("Float", "Distance 4","distance4")
+                self.newOutput("Vector", "Point 1","point1")
+                self.newOutput("Vector", "Point 2","point2")
+                self.newOutput("Vector", "Point 3","point3")
+                self.newOutput("Vector", "Point 4","point4")        
 
     def draw(self, layout):
-        self.drawNoiseSettings(layout)
+        layout.prop(self, "noiseSelect")
+        if self.noiseSelect == "FASTNOISE":
+            self.drawNoiseSettings(layout)
+        else:
+            if self.blenderNoiseType == "NOISE":
+                layout.prop(self, "noiseMode", text = "")
+                layout.prop(self, "noiseBasis", text = "")
+            elif self.blenderNoiseType == "3DNOISE":
+                layout.prop(self, "noiseBasis", text = "")
+            elif self.blenderNoiseType == "VARIABLELACUNARITY":
+                layout.prop(self, "noiseBasis", text = "")
+                layout.prop(self, "noiseBasis2", text = "")
+            elif self.blenderNoiseType == "VORONOI":
+                layout.prop(self, "voronoiDistanceMetric", text = "")    
 
     def drawAdvanced(self, layout):
-        self.drawAdvancedNoiseSettings(layout)
+        if self.noiseSelect == "FASTNOISE":
+            self.drawAdvancedNoiseSettings(layout)
+        else:
+            layout.prop(self, "blenderNoiseType")
+
+    def getExecutionFunctionName(self):
+        if self.noiseSelect == "FASTNOISE":
+            return "execute_FastNoise"
+        else:
+            if self.blenderNoiseType == "NOISE":
+                if self.noiseMode == "FRACTAL":
+                    return "execute_Fractal"
+                elif self.noiseMode == "MULTI_FRACTAL":
+                    return "execute_MultiFractal"
+                elif self.noiseMode == "HETERO_TERRAIN":
+                    return "execute_HeteroTerrain"
+                elif self.noiseMode == "RIDGED_MULTI_FRACTAL":
+                    return "execute_RigidMultiFractal"
+                elif self.noiseMode == "HYBRID_MULTI_FRACTAL":
+                    return "execute_HybridMultiFractal"
+            elif self.blenderNoiseType == "3DNOISE":
+                return "execute_3dNoise"
+            elif self.blenderNoiseType == "VARIABLELACUNARITY":
+                return "execute_VariableLacunarity"
+            elif self.blenderNoiseType == "VORONOI":
+                return "execute_Voronoi"    
+
+
 
-    def execute(self, vectors, *settings):
+    def execute_FastNoise(self, vectors, *settings):
         noise = self.calculateNoise(vectors, *settings)
         return DoubleList.fromValues(noise)
+
+    #Fractal Noise Methods:
+
+    def execute_Fractal(self, vectors, fractalDimension, lacunarity ,octaves):
+        if not self.useVectorList: vectors = Vector3DList.fromValue(vectors)
+        noiseBasis = self.noiseBasis
+        values = blFractal(noiseBasis, vectors, fractalDimension, lacunarity ,octaves)
```

Comment:
Change this `lacunarity ,octaves --> lacunarity, octaves`, and same for other places.